packages/node/src/backend.ts

```diff
@@ -24,79 +16,15 @@ export class NodeBackend extends BaseBackend<NodeOptions> {
    * @inheritDoc
    */
   // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/explicit-module-boundary-types
-  public eventFromException(exception: any, hint?: EventHint): PromiseLike<Event> {
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    let ex: any = exception;
-    const providedMechanism: Mechanism | undefined =
-      hint && hint.data && (hint.data as { mechanism: Mechanism }).mechanism;
-    const mechanism: Mechanism = providedMechanism || {
-      handled: true,
-      type: 'generic',
-    };
-
-    if (!isError(exception)) {
-      if (isPlainObject(exception)) {
-        // This will allow us to group events based on top-level keys
-        // which is much better than creating new group when any key/value change
-        const message = `Non-Error exception captured with keys: ${extractExceptionKeysForMessage(exception)}`;
-
-        getCurrentHub().configureScope(scope => {
-          scope.setExtra('__serialized__', normalizeToSize(exception as Record<string, unknown>));
-        });
-
-        ex = (hint && hint.syntheticException) || new Error(message);
-        (ex as Error).message = message;
-      } else {
-        // This handles when someone does: `throw "something awesome";`
-        // We use synthesized Error here so we can extract a (rough) stack trace.
-        ex = (hint && hint.syntheticException) || new Error(exception as string);
-        (ex as Error).message = exception;
-      }
-      mechanism.synthetic = true;
-    }
-
-    return new SyncPromise<Event>((resolve, reject) =>
-      parseError(ex as Error, this._options)
-        .then(event => {
-          addExceptionTypeValue(event, undefined, undefined);
-          addExceptionMechanism(event, mechanism);
-
-          resolve({
-            ...event,
-            event_id: hint && hint.event_id,
-          });
-        })
-        .then(null, reject),
-    );
+  public eventFromException(exception: any, hint?: EventHint | undefined): PromiseLike<Event> {
```

Comment:
Why does it need `| undefined` if it's already optional?

Reply:
> Would it maybe be possible to just bail out if `fs` is not available?

Any usage of `require('fs')` would be picked up by bundlers and throw errors which would force everyone to add `fs` to `externals`.